sdk/storage/Azure.Storage.Blobs/src/BlobUriBuilder.cs

```diff
@@ -4,99 +4,163 @@
         /// of a server.
+        /// 
         /// Example: "account.blob.core.windows.net"
         /// </summary>
-        public string Host;
+        public string Host
+        {
+            get => this._host;
+            set { this.ResetUri(); this._host = value; }
+        }
+        private string _host;
 
         /// <summary>
         /// Gets or sets the port number of the URI.
         /// </summary>
-        public int Port;
+        public int Port
+        {
+            get => this._port;
+            set { this.ResetUri(); this._port = value; }
+        }
+        private int _port;
 
         /// <summary>
         /// Gets or sets the Azure Storage account name.  This is only
-        /// populated for IP-style <see cref="Uri"/>s.
+        /// populated for IP-style <see cref="System.Uri"/>s.
         /// </summary>
-        public string AccountName;
+        public string AccountName
+        {
+            get => this._accountName;
+            set { this.ResetUri(); this._accountName = value; }
+        }
+        private string _accountName;
 
         /// <summary>
         /// Gets or sets the name of a blob storage Container.  The value
         /// defaults to <see cref="String.Empty"/> if not present in the
-        /// <see cref="Uri"/>.
+        /// <see cref="System.Uri"/>.
         /// </summary>
-        public string ContainerName;
+        public string ContainerName
+        {
+            get => this._containerName;
+            set { this.ResetUri(); this._containerName = value; }
+        }
+        private string _containerName;
 
         /// <summary>
         /// Gets or sets the name of a blob.  The value defaults to
-        /// <see cref="String.Empty"/> if not present in the <see cref="Uri"/>.
+        /// <see cref="String.Empty"/> if not present in the <see cref="System.Uri"/>.
         /// </summary>
-        public string BlobName;
+        public string BlobName
+        {
+            get => this._blobName;
+            set { this.ResetUri(); this._blobName = value; }
+        }
+        private string _blobName;
 
         /// <summary>
         /// Gets or sets the name of a blob snapshot.  The value defaults to
-        /// <see cref="String.Empty"/> if not present in the <see cref="Uri"/>.
+        /// <see cref="String.Empty"/> if not present in the <see cref="System.Uri"/>.
         /// </summary>
-        public string Snapshot;
+        public string Snapshot
+        {
+            get => this._snapshot;
+            set { this.ResetUri(); this._snapshot = value; }
+        }
+        private string _snapshot;
 
         ///// <summary>
-        ///// VersionId.  Empty string if not present in URI.
+        ///// Gets or sets the VersionId.  The value defaults to
+        ///// <see cref="String.Empty"/> if not present in the <see cref="Uri"/>.
         ///// </summary>
-        //public string VersionId;
+        //public string VersionId
```

Comment:
Why is this one commented out?

Reply:
It's a feature we don't fully support yet.  It was commented out in the private UriBuilders so I updated the comments to make it easier for folks to add back when they're ready.